inference-engine/samples/benchmark_app/main.cpp

```diff
@@ -325,7 +331,30 @@ int main(int argc, char *argv[]) {
             slog::info << "Loading network files" << slog::endl;
 
             auto startTime = Time::now();
-            CNNNetwork cnnNetwork = ie.ReadNetwork(FLAGS_m);
+
+            // Read XML file content into a string
+            std::ifstream xml_file(FLAGS_m);
+            std::string xml_content((std::istreambuf_iterator<char>(xml_file)), std::istreambuf_iterator<char>());
+#if defined(ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+            std::wstring bin_file = std::wstring_convert<std::codecvt_utf8<wchar_t>>().from_bytes(FLAGS_m);
+            bin_file = bin_file.substr(0, FLAGS_m.find_last_of('.')) + L".bin";
+#else
+            std::string bin_file = FLAGS_m.substr(0, FLAGS_m.find_last_of('.')) + ".bin";
+#endif
+
+            // Get file size, size is needed to create a tensor descriptor
+            std::ifstream bin_stream(bin_file, std::ios::in);
+            bin_stream.seekg(0, std::ios::end);
+            size_t bin_size = bin_stream.tellg();
+            bin_stream.close();
+
+            // Create allocator and blob with model data
+            std::shared_ptr<IAllocator> mmap = details::make_mmap_allocator(bin_file);
+            TensorDesc bin_td(Precision::U8, { bin_size }, Layout::C);
+            Blob::Ptr bin_blob = make_shared_blob<uint8_t>(bin_td, mmap);
+            bin_blob->allocate();
+
+            CNNNetwork cnnNetwork = ie.ReadNetwork(xml_content, bin_blob);
```

Comment:
I have 2 points:

1. benchmark_app read network time are measured by performance validation team. It would be nice if this change doesn't affect default behavior.

2. It brings misalignment of python and C++ tools behaviour.

 Can we made this feature optional in C++ benchmark_app and regulate via command  line parameter?

Reply:
Make sense, i can add one more switch, like "-mmap" which  will mean use mmap allocator  to read model params